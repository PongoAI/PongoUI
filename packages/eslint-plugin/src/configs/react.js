/** @type {import("eslint").Linter.Config} */
const config = {
  extends: ['airbnb-typescript', 'prettier'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.json',
  },
  plugins: ['deprecation', 'import', '@typescript-eslint', 'jest', 'jsx-a11y', 'react', 'react-hooks'],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {
        alwaysTryTypes: true,
      },
    },
    'no-unused-expressions': 'off',
  },
  env: {
    browser: true,
    es6: true,
    node: true,
  },

  ignorePatterns: ['coverage', 'dist', 'dist-storybook', 'lib', 'lib-commonjs', '**/__snapshots__', 'node_modules'],
  rules: {
    'no-unused-vars': 'error',
    'no-use-before-define': 'off',
    '@typescript-eslint/no-use-before-define': ['error'],
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
        ts: 'never',
        tsx: 'never',
      },
    ],
  },
};

module.exports = config;
