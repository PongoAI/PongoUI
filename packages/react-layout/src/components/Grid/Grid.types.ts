import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';

export type GridSlots = {
  /**
   * The root of the Grid.
   */
  root: Slot<'div'>;
};

export interface GridCommons {
  /**
   * The appearance of the Grid.
   * Focused has a lower padding and gap size then relaxed.
   * Relaxed should be used when the content of your app is minimal.
   *
   * @defaultValue focused
   */
  appearance?: 'focused' | 'relaxed';

  /**
   * How the components should be aligned horizontally.
   */
  horizontalAlignment?: 'start' | 'center' | 'end';

  /**
   * How the component should be aligned vertically.
   */
  verticalAlignment?: 'start' | 'center' | 'end';

  /**
   * The size of each item in the grid.
   */
  contentSize: string;
}

export type GridProps = ComponentProps<GridSlots> & GridCommons;

export type GridState = ComponentState<GridSlots> & GridCommons;
