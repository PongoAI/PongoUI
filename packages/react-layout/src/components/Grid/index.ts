export * from './Grid';
export * from './useGridStyles';
export * from './Grid.types';
export * from './useGrid';
