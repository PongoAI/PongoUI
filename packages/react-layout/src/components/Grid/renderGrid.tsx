import * as React from 'react';
import { getSlots } from '@fluentui/react-utilities';
import type { GridState, GridSlots } from './Grid.types';

export const renderGrid = (state: GridState) => {
  const { slots, slotProps } = getSlots<GridSlots>(state);

  return <slots.root {...slotProps.root} />;
};
