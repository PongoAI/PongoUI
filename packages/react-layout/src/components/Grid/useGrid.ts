import * as React from 'react';
import { getNativeElementProps } from '@fluentui/react-utilities';
import type { GridProps, GridState } from './Grid.types';

export const useGrid = (props: GridProps, ref: React.Ref<HTMLElement>): GridState => {
  const {
    appearance = 'focused',
    horizontalAlignment = 'center',
    verticalAlignment = 'center',
    contentSize = '0px',
  } = props;

  const state: GridState = {
    appearance,
    horizontalAlignment,
    verticalAlignment,
    contentSize,
    components: {
      root: 'div',
    },
    root: getNativeElementProps('div', {
      ref,
      ...props,
    }),
  };

  return state;
};
