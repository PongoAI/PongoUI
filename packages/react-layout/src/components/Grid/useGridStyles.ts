import { makeStyles, mergeClasses } from '@griffel/react';
import type { GridState } from './Grid.types';
import { tokens } from '@cebus/react-theme';

const gridContentWidth = `--cebus-grid-content-width`;

const useStyles = makeStyles({
  root: {
    display: 'grid',
    gridTemplateColumns: `repeat(auto-fit, minmax(var(${gridContentWidth}), max-content))`,
    width: '100%',
  },

  focused: {
    gridGap: tokens.focusedLayout,
  },

  relaxed: {
    gridGap: tokens.relaxedLayout,
  },

  centerHorizontal: {
    justifyContent: 'center',
  },

  startHorizontal: {
    justifyContent: 'start',
  },

  endHorizontal: {
    justifyContent: 'end',
  },

  centerVertical: {
    alignItems: 'center',
  },

  startVertical: {
    alignItems: 'start',
  },

  endVertical: {
    alignItems: 'end',
  },
});

export const useGridStyles = (state: GridState) => {
  const styles = useStyles();

  const CSSVariables = {
    [gridContentWidth]: state.contentSize,
  };

  state.root.className = mergeClasses(
    styles.root,
    styles[state.appearance!],
    styles[(state.horizontalAlignment! + 'Horizontal') as 'endHorizontal' | 'startHorizontal' | 'centerHorizontal'],
    styles[(state.verticalAlignment! + 'Vertical') as 'endVertical' | 'startVertical' | 'centerVertical'],
    state.root.className,
  );

  state.root.style = {
    ...CSSVariables,
    ...state.root.style,
  };

  return state;
};
