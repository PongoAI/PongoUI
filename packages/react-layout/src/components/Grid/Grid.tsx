import * as React from 'react';
import { useGrid } from './useGrid';
import { useGridStyles } from './useGridStyles';
import { renderGrid } from './renderGrid';
import type { GridProps } from './Grid.types';
import type { ForwardRefComponent } from '@fluentui/react-utilities';

/**
 * The Grid helps handle the positioning of a collection of items.
 */
export const Grid: ForwardRefComponent<GridProps> = React.forwardRef((props, ref) => {
  const state = useGrid(props, ref);

  useGridStyles(state);

  return renderGrid(state);
});
Grid.displayName = 'Grid';
