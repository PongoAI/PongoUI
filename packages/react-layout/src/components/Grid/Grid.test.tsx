import React from 'react';
import { Grid } from './Grid';
import { isConformant } from '../../common/isConformant';
import { render } from '@testing-library/react';

describe('Grid', () => {
  isConformant({
    Component: Grid,
    displayName: 'Grid',
  });

  it('renders a basic Grid', () => {
    const renderedComponent = render(<Grid contentSize="0px" />);

    expect(renderedComponent.baseElement).toMatchSnapshot();
  });
});
