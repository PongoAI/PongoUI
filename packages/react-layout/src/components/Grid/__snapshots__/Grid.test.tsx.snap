// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders a basic Grid 1`] = `
<body>
  <div>
    <div
      class="___131c3ln f13qh94s f8s1hii fly5x3f fnigl43 f4d9j23 f122n59"
      style="--cebus-grid-content-width: 0px;"
    />
  </div>
</body>
`;
